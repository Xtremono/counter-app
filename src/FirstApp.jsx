export const FirstApp = () => {
  return (
    <>
    <h1>Soy Tomas</h1>
    <p>Estoy tratando de conseguir mi primer trabajo en el mundo tech</p>
    </>
  );
};

// <></> A esto se le llama fragmento, es un agrupador de elementos html. Esto tambien lo conocemos como "Nodo Padre"
