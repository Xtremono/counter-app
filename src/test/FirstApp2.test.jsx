import { render, screen } from "@testing-library/react"
import { FirstApp } from "../FirstApp"




describe('FirstApp', () => {

    const title = "Hola, soy Goku"
    const subTitle = "Hola, soy Goku"

    test('hacer match con el snapshot', () => {

        const { container } = render(<FirstApp title={title}/>);
        expect(container).toMatchSnapshot();




    })

    test('debe demostar un mensaje que diga "Hola, soy Goku"', () => {

      render(<FirstApp title={title}/>);
      expect( screen.getByText(title)).toBeTruthy();



    })

    test('debe de mostrar el titulo de un h1', () => {

      render(<FirstApp title={title}/>);
      expect(screen.getByRole('heading', { level: 1}).innerHTML).toContain(title);

    })

    test('debe de enviar el subititulo enviado por props', () => {

      render(<FirstApp
        title={title}
        subTitle={subTitle}/>);

        expect( screen.getAllByText(subTitle).length).toBe(2);

    })
  })
