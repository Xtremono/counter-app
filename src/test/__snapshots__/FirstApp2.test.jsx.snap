// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  />
  <p>
    Hola, soy Goku
  </p>
  <p />
  <p />
</div>
`;
